src/components/SigningOfficialDAAPopup.js

```diff
@@ -0,0 +1,107 @@
+import { h, h2, a, p, div, span } from 'react-hyperscript-helpers';
+import Modal from 'react-modal';
+import { Notifications } from '../libs/utils';
+import BroadLibraryCardAgreementLink from '../assets/Library_Card_Agreement_2021.pdf';
+import NIHLibraryCardAgreementLink from '../assets/NIH_Library_Card_Agreement_11_17_22_version.pdf';
+
+import {Styles} from '../libs/theme';
+import { User } from '../libs/ajax';
+import Acknowledgments, { hasSOAcceptedDAAs } from '../libs/acknowledgements';
+import { useEffect, useState } from 'react';
+import { spinnerService } from '../libs/spinner-service';
+
+
+const acceptDaas = async () => {
+  return await User.acceptAcknowledgments(Acknowledgments.broadLcaAcknowledgement, Acknowledgments.nihLcaAcknowledgement);
+};
+
+export const SigningOfficialDAAPopup = () => {
+
+  const [hasAccepted, setHasAccepted] = useState();
+  const [isLoading, setIsLoading] = useState(false);
+
+  useEffect(() => {
+    const init = async() => {
+      try {
+        setIsLoading(true);
+        setHasAccepted(await hasSOAcceptedDAAs());
+        setIsLoading(false);
+      } catch(error) {
+        Notifications.showError({text: 'Error: Unable to retrieve current user from server'});
+        setIsLoading(false);
+      }
+    };
+    init();
+  }, []);
+
+  const acceptAndClose = () => {
+    acceptDaas().then(() => {
+      setHasAccepted(true);
+    }).catch(() => {
+      Notifications.showError({text: 'Failed to accept DAA agreements.'});
+      setHasAccepted(false);
+    });
+  };
+
+  useEffect(() => {
+    if (isLoading) {
+      spinnerService.showAll();
+    } else {
+      spinnerService.hideAll();
+    }
+  }, [isLoading]);
+
+  return h(Modal, {
+    isOpen: hasAccepted === false,
+    onRequestClose: acceptAndClose,
```

Comment:
This is why closing the modal triggers acceptance. We should only call `acceptAndClose` when the "I Agree" button is clicked.